### PR TITLE
Integrate Cookbook, Inventory and Cart into CookingPapa

### PR DESCRIPTION
**Pull request by gohkh**
Changes:
- Add Cookbook, Inventory, and Cart to Model, Storage, and Logic
- Update dependencies
- Update tests

The code runs pretty much the same as before, except that I have removed AddressBook from Storage completely, since we are not using the address book storage anymore. Model and Logic still contain references to AddressBook.

Gradle build is successful and all tests still pass. Tests for new classes have not been written.

### CODE REVIEW

File: src/main/java/seedu/address/model/Inventory.java [hunk 1]
```diff
@@ -66,8 +82,8 @@ public void removeIngredient(Ingredient key) {
 
     @Override
     public String toString() {
-        // TODO: After implementing Ingredients
-        return "";
+        return ingredients.asUnmodifiableObservableList().size() + " ingredients";
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/model/Inventory.java]: is the toString() only used to print out the size?

**Review by hans555 — APPROVED** [review]
Looks good for the time being

File: src/main/java/seedu/address/model/Inventory.java (hunk 1 above)
Comment (gohkh) [reply src/main/java/seedu/address/model/Inventory.java]: It's just copying the addressbook toString() style. Should we change it to display "Inventory: " + ingredients? (though on the user end this probably won't matter)

**Comment by gohkh** [comment]
Currently abstracting out common code in Inventory and Cart to an abstract base class

**Review by shanecsj — APPROVED** [review]
LGTM. We can proceed significantly from this PR.

**Review by teo-jun-xiong — APPROVED** [review]

**Review by tzihiang — APPROVED** [review]

**Pull request merged**